File: packages/core/src/types/bay.ts
export interface Bay {
  id: string;
  name: string;
  projectPath: string;
  gitBranch: string | null;
  status: BayStatus;
  windowState: WindowState;
  createdAt: string;
  updatedAt: string;
  lastAccessedAt: string;
}

export const BAY_STATUSES = ['active', 'inactive', 'loading', 'error'] as const;
export type BayStatus = (typeof BAY_STATUSES)[number];

export function isBayStatus(value: string): value is BayStatus {
  return (BAY_STATUSES as readonly string[]).includes(value);
}

export interface WindowState {
  x: number;
  y: number;
  width: number;
  height: number;
  isMaximized: boolean;
  leftRailWidth: number;
  rightRailWidth: number;
  bottomTrayHeight: number;
  openTabs: string[];
  activeTab: string | null;
}

File: apps/desktop/src-tauri/src/models/bay.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct Bay {
    pub id: String,
    pub name: String,
    pub project_path: String,
    pub git_branch: Option<String>,
    pub status: String,
    pub window_state: Option<String>,
    pub created_at: String,
    pub updated_at: String,
    pub last_accessed_at: String,
}
